File: app/src/main/java/com/test/nutshell/data/db/Storage.java
package com.test.nutshell.data.db;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.google.gson.Gson;
import com.test.nutshell.data.model.Repo;
import com.test.nutshell.data.model.User;

import java.util.List;

import javax.inject.Inject;

public class Storage {
    @NonNull
    private RepoDatabase repoDatabase;
    @NonNull
    private SharedPreferncesHelper preferncesHelper;

    @NonNull
    private Gson gson;

    @Inject
    public Storage(@NonNull RepoDatabase repoDatabase,
                   @NonNull SharedPreferncesHelper preferncesHelper, @NonNull Gson gson) {
        this.repoDatabase = repoDatabase;
        this.preferncesHelper = preferncesHelper;
        this.gson = gson;
    }

    @NonNull
    public User getUser() {
        return preferncesHelper.getUser();
    }

    public void putUser(@Nullable User user) {
        if (user != null) {
            preferncesHelper.putUser(user);
        } else {
            preferncesHelper.putUser(new User("", ""));
        }
    }

    @NonNull
    public List<Repo> getRepos() {
        return repoDatabase.repoDao().getAll();
    }

    public void putRepos(@NonNull List<Repo> repos) {
        repoDatabase.repoDao().insertAll(repos.toArray(new Repo[repos.size()]));
    }

    public void clearRepos() {
        repoDatabase.repoDao().clearTable();

    }

    @Nullable
    public String getQuery() {
        return preferncesHelper.getQuery();
    }

    public void putQuery(@NonNull String query) {
        preferncesHelper.setQuery(query);
    }


}

File: app/src/main/java/com/test/nutshell/data/db/Storage.kt
package com.test.nutshell.data.db

import android.content.ContentValues
import android.content.SharedPreferences
import com.google.gson.Gson
import com.test.nutshell.data.db.Db.Repo.toContentValues
import com.test.nutshell.data.db.Db.User.toContentValues
import com.test.nutshell.data.model.Repo
import com.test.nutshell.data.model.User
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton


@Singleton
class Storage
@Inject
constructor(private val dbOpenHelper: DbOpenHelper, private val gson: Gson, private val preferences: SharedPreferncesHelper) {

    fun getUser(): User? {
        val db = dbOpenHelper.readableDatabase

        db.query(Db.User.TABLE_NAME, Db.User.PROJECTION,
                null, null, null, null,
                Db.User.COL_USER_ID
        ).use { cursor ->
            val users = ArrayList<User>(cursor.count)
            if (cursor.count == 1) {
                cursor.moveToNext()
                return Db.User.parseCursor(cursor)
            }
            return null
        }
    }

    fun putUser(user: User?) {
        val db = dbOpenHelper.writableDatabase
        db.beginTransaction()

        try {
            db.delete(Db.User.TABLE_NAME, null, null)
            user?.let {
                val cv = ContentValues()
                user.toContentValues(cv)
                Db.insertOrReplace(db, Db.User.TABLE_NAME, cv)
            }
            db.setTransactionSuccessful()
        } finally {
            db.endTransaction()
        }
    }

    fun getRepos(): List<Repo> {
        val db = dbOpenHelper.readableDatabase

        db.query(Db.Repo.TABLE_NAME, Db.Repo.PROJECTION,
                null, null, null, null,
                Db.Repo.COL_ID
        ).use { cursor ->
            val photos = ArrayList<Repo>(cursor.count)

            while (cursor.moveToNext()) {
                val photo = Db.Repo.parseCursor(cursor)
                photos.add(photo)
            }

            return photos
        }
    }

    fun putRepos(repos: List<Repo>) {
        val db = dbOpenHelper.writableDatabase
        db.beginTransaction()

        try {
           // db.delete(Db.Repo.TABLE_NAME, null, null)

            val cv = ContentValues()

            for (photo in repos) {
                photo.toContentValues(cv)
                Db.insert(db, Db.Repo.TABLE_NAME, cv)
            }

            db.setTransactionSuccessful()
        } finally {
            db.endTransaction()
        }
    }
    fun clearRepos() {
        val db = dbOpenHelper.writableDatabase
        db.beginTransaction()

        try {
            db.delete(Db.Repo.TABLE_NAME, null, null)
            db.setTransactionSuccessful()
        } finally {
            db.endTransaction()
        }
    }

    fun getQuery():String? {
        return preferences.query
    }
    fun putQuery(query: String) {
        preferences.setQuery(query)
    }


}
